doc/api/fs.md

```diff
@@ -1101,11 +1101,62 @@ changes:
 * Returns: {AsyncIterator} An AsyncIterator that yields the paths of files
   that match the pattern.
 
+Retrieves the files matching the specified pattern(s).
+
+#### Pattern Syntax
+
+The glob implementation supports the following pattern syntax:
+
+**Basic Wildcards:**
+* `*` - Matches any number of characters within a single path segment
+* `?` - Matches exactly one character
+* `[abc]` - Matches any character in the brackets
+* `[a-z]` - Matches any character in the range
+* `[!abc]` or `[^abc]` - Matches any character not in the brackets
+
+**Globstar:**
+* `**` - Matches zero or more directories recursively
+* `**/*.js` - Matches all `.js` files in any subdirectory
+
+**Brace Expansion:**
+* `{a,b,c}` - Matches any of the comma-separated patterns
+* `{1..5}` - Matches any number in the range (1, 2, 3, 4, 5)
+* `*.{js,ts}` - Matches files with `.js` or `.ts` extensions
+
+**Extended Glob Patterns:**
+* `+(pattern)` - Matches one or more occurrences of the pattern
+* `*(pattern)` - Matches zero or more occurrences of the pattern
+* `?(pattern)` - Matches zero or one occurrence of the pattern
+* `!(pattern)` - Matches anything except the pattern
+
+#### Examples
+
 ```mjs
 import { glob } from 'node:fs/promises';
 
-for await (const entry of glob('**/*.js'))
-  console.log(entry);
+// Basic patterns
+for await (const entry of glob('*.js'))
+  console.log(entry); // All .js files in current directory
+
+// Recursive search
+for await (const entry of glob('**/*.{js,ts}'))
+  console.log(entry); // All .js and .ts files in any subdirectory
+
+// Brace expansion
+for await (const entry of glob('src/{components,utils}/**/*.js'))
+  console.log(entry); // .js files in src/components or src/utils
+
+// Extended glob patterns
+for await (const entry of glob('test/**/*.+(spec|test).js'))
+  console.log(entry); // Test files ending with .spec.js or .test.js
```

Comment:
This will match `foo.spectest.js`, `bar.testtesttesttestspecttest.js`, etc.
Seems like a bad example of this pattern type?